### ISSUE TITLE
[Bug]: p2c crash caused an infinite loop in the exception handler

### ISSUE THREAD

**Issue by hackerftsg** [Yimura/YimMenu#1001]
### Describe the bug

I asked a friend to use his p2c (x-force) to crash me while having YimMenu injected out of curiosity and it happened to cause an infinite loop in the exception handler logging the same error before the actual crash. The crash appears to occur in the receive_net_message hook as shown in the log

### Steps To Reproduce

crash

### Expected Behavior

not crash

### Screenshots and/or Logs

```
EXCEPTION_ACCESS_VIOLATION
Dumping modules:
GTA5.exe Base Address: 0x7FF70C050000 Size: 63296000
ntdll.dll Base Address: 0x7FFD8BDF0000 Size: 1245184
KERNEL32.DLL Base Address: 0x7FFD8AD90000 Size: 528384
KERNELBASE.dll Base Address: 0x7FFD89620000 Size: 1609728
USER32.dll Base Address: 0x7FFD8B070000 Size: 606208
win32u.dll Base Address: 0x7FFD895F0000 Size: 49152
GDI32.dll Base Address: 0x7FFD8AFE0000 Size: 57344
gdi32full.dll Base Address: 0x7FFD89460000 Size: 655360
msvcp_win.dll Base Address: 0x7FFD89B60000 Size: 335872
ucrtbase.dll Base Address: 0x7FFD899C0000 Size: 798720
PSAPI.DLL Base Address: 0x7FFD8AF80000 Size: 4096
WS2_32.dll Base Address: 0x7FFD8A980000 Size: 290816
RPCRT4.dll Base Address: 0x7FFD8B3E0000 Size: 876544
CRYPT32.dll Base Address: 0x7FFD89230000 Size: 1134592
WINTRUST.dll Base Address: 0x7FFD89580000 Size: 307200
msvcrt.dll Base Address: 0x7FFD89E60000 Size: 507904
IMM32.dll Base Address: 0x7FFD8B500000 Size: 131072
SHLWAPI.dll Base Address: 0x7FFD8B010000 Size: 212992
ADVAPI32.dll Base Address: 0x7FFD89F10000 Size: 425984
sechost.dll Base Address: 0x7FFD89C00000 Size: 446464
SHELL32.dll Base Address: 0x7FFD8A190000 Size: 6488064
ole32.dll Base Address: 0x7FFD89FC0000 Size: 897024
combase.dll Base Address: 0x7FFD8AA00000 Size: 2543616
OLEAUT32.dll Base Address: 0x7FFD8B690000 Size: 647168
VERSION.dll Base Address: 0x7FFD7ED80000 Size: 12288
GFSDK_ShadowLib.win64.dll Base Address: 0x7FFCD3220000 Size: 96256
DSOUND.dll Base Address: 0x7FFCED8D0000 Size: 491520
bink2w64.dll Base Address: 0x7FFCED840000 Size: 307712
WINMM.dll Base Address: 0x7FFD807A0000 Size: 114688
WININET.dll Base Address: 0x7FFD60D80000 Size: 2113536
MFPlat.DLL Base Address: 0x7FFD80BE0000 Size: 1503232
shcore.dll Base Address: 0x7FFD8B220000 Size: 679936
msdmo.dll Base Address: 0x7FFD826E0000 Size: 16384
MF.dll Base Address: 0x7FFD82220000 Size: 368640
PROPSYS.dll Base Address: 0x7FFD83DF0000 Size: 688128
MFReadWrite.dll Base Address: 0x7FFCF2870000 Size: 1003520
d3d9.dll Base Address: 0x7FFD748C0000 Size: 1306624
GFSDK_TXAA_AlphaResolve.win64.dll Base Address: 0x7FFD70550000 Size: 48128
WTSAPI32.dll Base Address: 0x7FFD85FC0000 Size: 40960
XINPUT1_3.dll Base Address: 0x400000 Size: 85504
bcrypt.dll Base Address: 0x7FFD88BF0000 Size: 110592
SETUPAPI.dll Base Address: 0x7FFD8B930000 Size: 937984
DINPUT8.dll Base Address: 0x7FFD628B0000 Size: 176128
IPHLPAPI.DLL Base Address: 0x7FFD87D90000 Size: 122880
powrprof.dll Base Address: 0x7FFD88110000 Size: 77824
winmmbase.dll Base Address: 0x7FFD6D980000 Size: 102400
d3dx9_43.dll Base Address: 0x7FFCD2EF0000 Size: 2280960
ResampleDmo.DLL Base Address: 0x7FFCD3160000 Size: 503808
cfgmgr32.dll Base Address: 0x7FFD88F20000 Size: 221184
dwmapi.dll Base Address: 0x7FFD865C0000 Size: 90112
OPENGL32.dll Base Address: 0x7FFD5B120000 Size: 798720
GLU32.dll Base Address: 0x7FFD5BCC0000 Size: 126976
dxcore.dll Base Address: 0x7FFD86630000 Size: 131072
D3DCOMPILER_43.dll Base Address: 0x7FFCD2C80000 Size: 2423296
MFCORE.DLL Base Address: 0x7FFD81780000 Size: 3252224
UMPDC.dll Base Address: 0x7FFD880F0000 Size: 36864
kernel.appcore.dll Base Address: 0x7FFD88340000 Size: 36864
bcryptPrimitives.dll Base Address: 0x7FFD89AE0000 Size: 380928
RTWorkQ.DLL Base Address: 0x7FFD82000000 Size: 102400
MSASN1.dll Base Address: 0x7FFD88A90000 Size: 36864
gameoverlayrenderer64.dll Base Address: 0x7FFCF29F0000 Size: 982528
inputhost.dll Base Address: 0x7FFD705F0000 Size: 1515520
CoreMessaging.dll Base Address: 0x7FFD85B10000 Size: 860160
CRYPTBASE.DLL Base Address: 0x7FFD88A50000 Size: 12288
windows.storage.dll Base Address: 0x7FFD87210000 Size: 6909952
wintypes.dll Base Address: 0x7FFD870D0000 Size: 544768
dbghelp.dll Base Address: 0x7FFD86BC0000 Size: 1626112
dbgcore.DLL Base Address: 0x7FFD7D570000 Size: 131072
uxtheme.dll Base Address: 0x7FFD86340000 Size: 425984
clbcatq.dll Base Address: 0x7FFD8B330000 Size: 471040
amsi.dll Base Address: 0x7FFD746D0000 Size: 57344
USERENV.dll Base Address: 0x7FFD888D0000 Size: 90112
profapi.dll Base Address: 0x7FFD89160000 Size: 73728
MpOav.dll Base Address: 0x7FFD74650000 Size: 204800
cryptnet.dll Base Address: 0x7FFD7ED40000 Size: 139264
drvstore.dll Base Address: 0x7FFD7EBE0000 Size: 1044480
devobj.dll Base Address: 0x7FFD88EE0000 Size: 135168
wldp.dll Base Address: 0x7FFD88B00000 Size: 184320
imagehlp.dll Base Address: 0x7FFD8A170000 Size: 49152
CRYPTSP.dll Base Address: 0x7FFD88A30000 Size: 57344
rsaenh.dll Base Address: 0x7FFD882B0000 Size: 147456
gpapi.dll Base Address: 0x7FFD88870000 Size: 77824
WINNSI.DLL Base Address: 0x7FFD865F0000 Size: 16384
NSI.dll Base Address: 0x7FFD89D50000 Size: 8192
steam_api64.dll Base Address: 0x529F0000 Size: 110592
steamclient64.dll Base Address: 0x7FFCF12C0000 Size: 15837184
vstdlib_s64.dll Base Address: 0x7FFCF3BB0000 Size: 285184
tier0_s64.dll Base Address: 0x7FFCF1120000 Size: 257536
Secur32.dll Base Address: 0x7FFD5AF20000 Size: 12288
MSWSOCK.dll Base Address: 0x7FFD887E0000 Size: 331776
SSPICLI.DLL Base Address: 0x7FFD885C0000 Size: 126976
D3D11.DLL Base Address: 0x7FFD84C00000 Size: 1867776
dxgi.dll Base Address: 0x7FFD864B0000 Size: 651264
directxdatabasehelper.dll Base Address: 0x7FFD82720000 Size: 196608
nvapi64.dll Base Address: 0x7FFD75A50000 Size: 6077952
D3D10_1.DLL Base Address: 0x7FFD5C2C0000 Size: 147456
d3d10_1core.dll Base Address: 0x7FFD704E0000 Size: 12288
MSCTF.dll Base Address: 0x7FFD8AE60000 Size: 917504
textinputframework.dll Base Address: 0x7FFD71030000 Size: 921600
WINSTA.dll Base Address: 0x7FFD87D20000 Size: 229376
nvldumdx.dll Base Address: 0x7FFCF1070000 Size: 545280
nvwgf2umx.dll Base Address: 0x7FFCEE720000 Size: 24333312
HID.DLL Base Address: 0x7FFD87AE0000 Size: 24576
MMDevApi.dll Base Address: 0x7FFD79C40000 Size: 462848
xaudio2_7.dll Base Address: 0x7FFCF6A20000 Size: 485376
AUDIOSES.DLL Base Address: 0x7FFD74C40000 Size: 1384448
resourcepolicyclient.dll Base Address: 0x7FFD869B0000 Size: 36864
Windows.UI.dll Base Address: 0x7FFD70870000 Size: 966656
avrt.dll Base Address: 0x7FFD82BD0000 Size: 8192
igd10iumd64.dll Base Address: 0x7FFD7D750000 Size: 12273664
ncrypt.dll Base Address: 0x7FFD88BC0000 Size: 114688
apphelp.dll Base Address: 0x7FFD85FE0000 Size: 335872
NTASN1.dll Base Address: 0x7FFD88B80000 Size: 69632
igdgmm64.dll Base Address: 0x7FFD7D210000 Size: 3204096
igc64.dll Base Address: 0x7FFD7A480000 Size: 23888384
napinsp.dll Base Address: 0x7FFD80A00000 Size: 53248
pnrpnsp.dll Base Address: 0x7FFD80910000 Size: 69632
DNSAPI.dll Base Address: 0x7FFD87DC0000 Size: 716800
winrnr.dll Base Address: 0x7FFD808F0000 Size: 32768
wshbth.dll Base Address: 0x7FFD7F720000 Size: 45056
nlansp_c.dll Base Address: 0x7FFD54AB0000 Size: 81920
fwpuclnt.dll Base Address: 0x7FFD802F0000 Size: 294912
rasadhlp.dll Base Address: 0x7FFD7EB80000 Size: 8192
socialclub.dll Base Address: 0x7FFCEDB40000 Size: 5077504
chrome_elf.dll Base Address: 0x7FFCEDA30000 Size: 685568
libcef.dll Base Address: 0x7FFCD49B0000 Size: 109755392
COMDLG32.dll Base Address: 0x7FFD89D60000 Size: 774144
NETAPI32.dll Base Address: 0x7FFD68480000 Size: 49152
OLEACC.dll Base Address: 0x7FFD5DED0000 Size: 274432
UIAutomationCore.DLL Base Address: 0x7FFD52060000 Size: 3047424
COMCTL32.dll Base Address: 0x7FFD5D870000 Size: 573440
DWrite.dll Base Address: 0x7FFD84990000 Size: 1605632
WINSPOOL.DRV Base Address: 0x7FFD61CC0000 Size: 360448
USP10.dll Base Address: 0x7FFD7EB40000 Size: 4096
dxva2.dll Base Address: 0x7FFD4F3E0000 Size: 102400
credui.dll Base Address: 0x7FFD70530000 Size: 65536
urlmon.dll Base Address: 0x7FFD6B130000 Size: 1298432
WINHTTP.dll Base Address: 0x7FFD82AA0000 Size: 958464
CRYPTUI.dll Base Address: 0x7FFD62470000 Size: 278528
dhcpcsvc.DLL Base Address: 0x7FFD82A80000 Size: 69632
wevtapi.dll Base Address: 0x7FFD84170000 Size: 245760
ESENT.dll Base Address: 0x7FFD600D0000 Size: 3112960
iertutil.dll Base Address: 0x7FFD6AE40000 Size: 557056
srvcli.dll Base Address: 0x7FFD6B100000 Size: 57344
netutils.dll Base Address: 0x7FFD87EC0000 Size: 20480
WKSCLI.DLL Base Address: 0x7FFD82990000 Size: 49152
SAMCLI.DLL Base Address: 0x7FFD83FD0000 Size: 61440
FirewallAPI.dll Base Address: 0x7FFD87C40000 Size: 380928
fwbase.dll Base Address: 0x7FFD87C00000 Size: 163840
FWPolicyIOMgr.dll Base Address: 0x7FFD6F0C0000 Size: 241664
wdmaud.drv Base Address: 0x7FFD68580000 Size: 200704
ksuser.dll Base Address: 0x7FFD856A0000 Size: 8192
msacm32.drv Base Address: 0x7FFD85690000 Size: 20480
MSACM32.dll Base Address: 0x7FFD7EBC0000 Size: 77824
midimap.dll Base Address: 0x7FFD85680000 Size: 16384
ondemandconnroutehelper.dll Base Address: 0x7FFD50280000 Size: 45056
dhcpcsvc6.DLL Base Address: 0x7FFD82BE0000 Size: 28672
wbemprox.dll Base Address: 0x7FFD7D560000 Size: 28672
wbemcomn.dll Base Address: 0x7FFD752D0000 Size: 311296
wbemsvc.dll Base Address: 0x7FFD79890000 Size: 8192
fastprox.dll Base Address: 0x7FFD74F60000 Size: 708608
CoreUIComponents.dll Base Address: 0x7FFD81B70000 Size: 2036736
YimMenu.dll Base Address: 0x7FFCE8400000 Size: 5390848
MSVCP140.dll Base Address: 0x7FFD6D8F0000 Size: 346112
VCRUNTIME140_1.dll Base Address: 0x7FFD703B0000 Size: 15360
VCRUNTIME140.dll Base Address: 0x7FFD6D8D0000 Size: 65024
D3DCOMPILER_47.dll Base Address: 0x7FFD842A0000 Size: 3043328
schannel.DLL Base Address: 0x7FFD881B0000 Size: 544768
ncryptsslp.dll Base Address: 0x7FFD5BD10000 Size: 86016
xinput1_4.dll Base Address: 0x7FFD7EB60000 Size: 32768
Dumping registers:
RAX: 0x21E16E4FA10
RCX: 0xFFFFFFFFFFFFFFFF
RDX: 0x7FF70DF68190
RBX: 0x21E11C2C468
RSI: 0x21E11C2C240
RDI: 0x21E16E4FA10
RSP: 0x580B0FC1B0
RBP: 0x580B0FC250
R8:  0x6A
R9:  0x6B
R10: 0x10
R11: 0x213CD2
R12: 0x21E11C2C240
R13: 0x1
R14: 0x0
R15: 0x1
Dumping stacktrace:
[0]	GTA5.exe+0x620BBD 0x7FF70C670BBD
[1]	GTA5.exe+0x5F4B7C 0x7FF70C644B7C
[2]	GTA5.exe+0x7B8E0D 0x7FF70C808E0D
[3]	GTA5.exe+0x7B8EE8 0x7FF70C808EE8
[4]	GTA5.exe+0x72BCDD 0x7FF70C77BCDD
[5]	GTA5.exe+0x72CD68 0x7FF70C77CD68
[6]	GTA5.exe+0x6F1AF7 0x7FF70C741AF7
[7]	GTA5.exe+0x111E4E5 0x7FF70D16E4E5
[8]	GTA5.exe+0x169A5A1 0x7FF70D6EA5A1
[9]	GTA5.exe+0x10C973D 0x7FF70D11973D
[10]	GTA5.exe+0x1102EE8 0x7FF70D152EE8
[11]	GTA5.exe+0x16ADAC8 0x7FF70D6FDAC8
[12]	GTA5.exe+0x16A5560 0x7FF70D6F5560
[13]	GTA5.exe+0x16AABC0 0x7FF70D6FABC0
[14]	GTA5.exe+0x13E0939 0x7FF70D430939
[15]	GTA5.exe+0x16AACED 0x7FF70D6FACED
[16]	GTA5.exe+0x142D76E 0x7FF70D47D76E
[17]	C:\Users\userpc\Documents\C++\YimMenu\src\hooks\protections\receive_net_message.cpp L: 395 big::hooks::receive_net_message
[18]	GTA5.exe+0x142D821 0x7FF70D47D821
[19]	GTA5.exe+0x142D53A 0x7FF70D47D53A
[20]	GTA5.exe+0x1457A96 0x7FF70D4A7A96
[21]	GTA5.exe+0x10AA500 0x7FF70D0FA500
[22]	GTA5.exe+0x16056BD 0x7FF70D6556BD
[23]	GTA5.exe+0x1604EEF 0x7FF70D654EEF
[24]	GTA5.exe+0x2778E 0x7FF70C07778E
[25]	GTA5.exe+0x20EA3 0x7FF70C070EA3
[26]	GTA5.exe+0x1604AF8 0x7FF70D654AF8
[27]	GTA5.exe+0x1494 0x7FF70C051494
[28]	GTA5.exe+0x12DFE13 0x7FF70D32FE13
[29]	GTA5.exe+0x17F7170 0x7FF70D847170
[30]	KERNEL32.DLL BaseThreadInitThunk
[31]	ntdll.dll RtlUserThreadStart
--------End of exception--------
```

### Store

Steam

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

**Comment by SkiddyToast** [comment]
Seems to go through the `nonPhysicalDataMsg` reader.

**Comment by SkiddyToast** [comment]
Also seems to go through the `CPedCreationDataNode`.
`GTA5.exe+0x111E4E5` calls the `CPedFactory::CreatePed` virtual function.
Looking more into it, it seems that `CPed->m_model_info->qword228 == 0xFFFFFFFFFFFFFFFF`

**Comment by JDMJay** [comment]
was the crash named "big chungas"

**Comment by Aure7138** [comment]
I believe "Basic Crash" is related to CPed models and even some vehicle models

**Comment by Yimura** [comment]
Possibly fixed with #1006
